backend/src/controller/chatController.ts

```diff
@@ -160,90 +220,119 @@ async function handleChatToGPT(req: OpenAiChatRequest, res: Response) {
 	if (message.length > MESSAGE_CHARACTER_LIMIT) {
 		handleChatError(
 			res,
-			chatResponse,
-			true,
+			initChatResponse,
 			'Message exceeds character limit',
 			400
 		);
 		return;
 	}
-
-	// keep track of the number of sent emails
-	const numSentEmails = req.session.levelState[currentLevel].sentEmails.length;
+	const totalSentEmails: EmailInfo[] = [
+		...req.session.levelState[currentLevel].sentEmails,
+	];
 
 	// use default model for levels, allow user to select in sandbox
 	const chatModel =
 		currentLevel === LEVEL_NAMES.SANDBOX
 			? req.session.chatModel
 			: defaultChatModel;
 
-	// record the history before chat completion called
-	const chatHistoryBefore = [
+	const currentChatHistory = [
 		...req.session.levelState[currentLevel].chatHistory,
 	];
+	const defences = [...req.session.levelState[currentLevel].defences];
+
+	let levelResult = null;
```

Comment:
Not allowed by linter! If we were allowed, what would the default value be?

Ideally though we'd have something like

```typescript
const chatHandler = currentLevel < LEVEL_NAMES.LEVEL_3 ?
  handleChatWithoutDefenceDetection :
  handleChatWithDefenceDetection;
const levelResult = chatHandler (...).catch((error) => {
  // handle error
})
``` 

but I'm scared to make more changes, so can we leave that for the future?

Reply:
Why does the linter complain, what's the rule violation? Let me check...

Ah, it's a rule we are explicitly turning on in our linter config:

```
'@typescript-eslint/init-declarations': 'error',
```

I wonder why. It's fairly common to set the value of a `let` conditionally, particularly when the logic is a little complex. Using ternary assignment is a possibility, but it can become a bit gnarly, particularly if there are try-catches involved. The compiler will complain anyway if a `let` defined in this way is used when it may not have been initialised, and a `const` must be declared with a value anyway.

I'd vote for removing this rule from our linter config. Frontend also, for consistency. Note it's not a rule that's included in any of the recommended rulesets we are using, so we can just remove the lines, instead of turning them off.